### PR TITLE
HUH-74: Interstitial page sign-in hint: Set session and redirect to IDP

### PR DESCRIPTION
**Pull request by Wynndow**
We're testing a new flow where a journey hint is set on the interstitial
page if it is determined the user has previously signed in successfully.
The hint is the simple_id of the IDP, prefixed with idp_.

These changes allow that hint to be picked up, and the idp selected,
before redirecting the user to the idp.

### CODE REVIEW

File: app/controllers/authn_request_controller.rb [hunk 1]
```diff
@@ -128,4 +131,8 @@ def store_rp_request_data
   def raise_error_if_params_invalid
     something_went_wrong_warn("Missing/empty SAML message from #{request.referer}", :bad_request) if params['SAMLRequest'].blank?
   end
+
+  def successful_idp_hint?
+    ->(hint) { hint && hint.start_with?('idp_') }
```
Comment (CharlesIC) [inline app/controllers/authn_request_controller.rb]: `->(hint)` is this correct Ruby syntax?

Comment (Wynndow) [reply app/controllers/authn_request_controller.rb]: It is.

Comment (CharlesIC) [reply app/controllers/authn_request_controller.rb]: Hmm that's right... https://www.rubyguides.com/2016/02/ruby-procs-and-lambdas/
I'm so used to lambdas being `x ​-> { f(x) }` that this looks weird to me

Comment (Wynndow) [reply app/controllers/authn_request_controller.rb]: Ruby's a weird language dude.

**Pull request merged**